core/main_test.ts

```diff
@@ -881,6 +881,11 @@ actions:
       const dataPreparationYaml = `
 nodes:
 - id: node1
+  unknown1: val1
+  unknown2: 1234
+  unknown3: 
+    nested1: 1234
+    nesteda: qwerty
```

Comment:
I don't think it's worth breaking proto safety to allow fields like this.

I think what you're looking for is adding a map instead:

```proto
message DataPreparation {
  ...
    map<string, SomeStructure> some_structure_map_name = 1;
}
```